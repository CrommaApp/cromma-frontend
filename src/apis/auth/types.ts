import { BasicResult } from '@apis/index';

type userInfo = {
	Keywords: string[];
	createdAt: string;
	id: number;
	updatedAt: string;
	userId: string;
};

export type SignRequest = {
	userId: string;
	password: string;
};
export type LoginResult = BasicResult & {
	data: userInfo;
};

export type getMyInfoResult = BasicResult & {
	data: userInfo | null;
};
