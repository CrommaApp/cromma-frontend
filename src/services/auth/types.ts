import { BasicResult } from '@services/api';
import { Post } from '@services/post/types';

type userInfo = {
	createdAt: string;
	id: number;
	updatedAt: string;
	userId: string;
};

export type SignRequest = {
	userId: string;
	password: string;
};
export type LoginResult = BasicResult & {
	data: userInfo;
};

export type getMyInfoResult = BasicResult & {
	data: userInfo | null;
};
