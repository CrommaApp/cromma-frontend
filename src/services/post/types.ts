import { BasicResult } from '@services/api';

export type Post = {
	id: number;
	title: string;
	content: string;
	createdAt: string;
	updatedAt: string;
};

export type GetAllPostsResult = BasicResult & {
	data: Post[] & { UserId: string };
};

export type UploadPostRequest = {
	title: string;
	content: string;
};

export type UploadPostResult = BasicResult & {
	data: Post;
};

export type getPostRequest = {
	postId: string;
};

export type deletePostRequest = {
	postId: string;
};
