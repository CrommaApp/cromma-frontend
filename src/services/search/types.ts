export type AddRecentKeywordRequest = {
	content: string;
};
