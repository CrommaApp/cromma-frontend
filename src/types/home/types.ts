export type Keyword = {
	id: number;
	content: string;
};
