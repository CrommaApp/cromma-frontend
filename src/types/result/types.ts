export type News = {
	sfty_notice_id: string;
	title: string;
	wrt_dt: string;
	txt_origin_cn: string;
};

export type NewsInfo = {
	continent_cd: null | string;
	continent_eng_nm: null | string;
	continent_nm: null | string;
	country_eng_nm: null | string;
	country_iso_alp2: null | string;
	country_nm: string;
	file_cnt: null | number;
	file_download_url: string;
	file_path: null | string;
	html_origin_cn: string;
	sfty_notice_id: string;
	title: string;
	txt_origin_cn: string;
	wrt_dt: string;
};

export type NewsData = {
	numOfRows: number;
	pageNo: number;
	resultCode: number;
	resultMsg: string;
	totalCount: number;
	data: NewsInfo[];
};
