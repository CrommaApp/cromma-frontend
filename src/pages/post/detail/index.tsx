import React, { useEffect } from 'react';
import PostDetailContent from '@components/features/detail/detail-content';
import { ContentWrapper } from '@components/shared/content-wrapper/styled';
import useGetPost from '@hooks/useGetPost';

const PostDetail = () => {
	const [post, getPost] = useGetPost();

	useEffect(() => {
		getPost();
	}, []);

	return (
		<>
			<ContentWrapper>
				<h1 className="a11y-hidden">게시글 전체 내용</h1>
				<PostDetailContent post={post} />
			</ContentWrapper>
		</>
	);
};

export default PostDetail;
