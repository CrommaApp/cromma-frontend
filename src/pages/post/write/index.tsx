import React from 'react';

const PostWrite = () => {
	return (
		<div>
			<>write</>
		</div>
	);
};

export default PostWrite;
