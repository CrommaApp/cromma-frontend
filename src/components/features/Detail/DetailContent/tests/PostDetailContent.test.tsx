import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import PostDetailContent from '@components/features/Detail/DetailContent';
import { Post } from '@services/post/types';
import { RecoilRoot } from 'recoil';

describe('PostDetailContent Component', () => {
	const mockPost: Post = {
		id: 1,
		title: 'test',
		content: 'lorem',
		createdAt: '2021-12-12',
		updatedAt: '2021-12-14',
		User: {
			userId: 'test@gmail.com',
		},
		UserId: 2,
	};

	const mockedDeletePost = jest.fn();

	it('Snapshot', () => {
		const component = renderer.create(
			<RecoilRoot>
				<PostDetailContent post={mockPost} isMyPost={true} deletePost={mockedDeletePost} />
			</RecoilRoot>,
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Call deletePost() when click delete button', () => {
		render(
			<RecoilRoot>
				<PostDetailContent post={mockPost} isMyPost={true} deletePost={mockedDeletePost} />
			</RecoilRoot>,
		);

		const deleteButton = screen.getByRole('button', {
			name: '삭제하기',
		});

		userEvent.click(deleteButton);

		expect(mockedDeletePost).toBeCalled();
	});

	it('Hide delete button if post is not mine', () => {
		render(
			<RecoilRoot>
				<PostDetailContent post={mockPost} isMyPost={false} deletePost={mockedDeletePost} />
			</RecoilRoot>,
		);

		expect(() =>
			screen.getByRole('button', {
				name: '삭제하기',
			}),
		).toThrowError();
	});
});
