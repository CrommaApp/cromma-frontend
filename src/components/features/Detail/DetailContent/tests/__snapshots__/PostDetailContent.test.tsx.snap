// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetailContent Component Snapshot 1`] = `
<article
  className="sc-bdvvtL eFZVCe"
>
  <h2>
    test
  </h2>
  <time
    dateTime="2021-12-12"
  >
    2021-12-12
  </time>
  <p>
    작성자 : 
    test@gmail.com
  </p>
  <p>
    lorem
  </p>
  <button
    onClick={[MockFunction]}
    type="button"
  >
    삭제하기
  </button>
</article>
`;
